OpenRA.Mods.Common/Lint/CheckTraitLocation.cs

```diff
@@ -0,0 +1,47 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2021 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Linq;
+using DiscordRPC.Helper;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Lint
+{
+	public class CheckTraitLocation : ILintRulesPass
+	{
+		void ILintRulesPass.Run(Action<string> emitError, Action<string> emitWarning, ModData modData, Ruleset rules)
+		{
+			foreach (var actorInfo in rules.Actors)
+			{
+				foreach (var traitInfo in actorInfo.Value.TraitInfos<TraitInfo>())
+				{
+					var traitLocation = traitInfo.GetType().GetCustomAttributes<TraitLocationAttribute>(true).FirstOrDefault();
+					if (traitLocation == null)
+						continue;
+
+					if (!Enum.TryParse(actorInfo.Key, true, out SystemActors systemActor))
+						continue;
```

Comment:
This prevents the check from working when a restricted trait is defined on any general actor (e.g. `1tnk`). We should be emitting an error here instead!